apps/astarte_housekeeping/lib/astarte_housekeeping/c_system.ex

```diff
@@ -46,61 +51,60 @@ defmodule CSystem do
           {:error, :timeout}
         else
           Process.sleep(millis)
-          wait_schema_agreement(conn, timeout - millis)
+          wait_schema_agreement(timeout - millis)
         end
-
-      any_other ->
-        any_other
     end
   end
 
-  def schema_versions(conn) do
-    with {:ok, local_version} <- query_local_schema_version(conn),
-         {:ok, peers_versions} <- query_peers_schema_versions(conn) do
+  def schema_versions() do
+    with {:ok, local_version} <- query_local_schema_version(),
+         {:ok, peers_versions} <- query_peers_schema_versions() do
       {:ok, Enum.uniq([local_version | peers_versions])}
     end
   end
 
-  def query_peers_schema_versions(conn) do
-    query = "SELECT schema_version FROM system.peers"
+  def query_peers_schema_versions() do
+    query =
+      from(peers in "peers",
+        select: peers.schema_version
+      )
 
-    with {:ok, res} <- Xandra.execute(conn, query, %{}, consistency: :one) do
-      schema_versions =
-        res
-        |> Stream.map(&Map.fetch!(&1, "schema_version"))
-        |> Stream.uniq()
-        |> Enum.to_list()
+    case Repo.all(query, consistency: :one) do
+      res ->
+        schema_versions =
+          res
+          |> Stream.uniq()
+          |> Enum.to_list()
 
-      {:ok, schema_versions}
+        {:ok, schema_versions}
     end
   end
 
-  def query_local_schema_version(conn) do
-    query = "SELECT schema_version FROM system.local WHERE key='local'"
-
-    with {:ok, res} <- Xandra.execute(conn, query, %{}, consistency: :one) do
-      schema_version =
-        res
-        |> Enum.take(1)
-        |> List.first()
-        |> Map.fetch!("schema_version")
+  def query_local_schema_version() do
+    query =
+      from(locals in "locals",
+        where: locals.key == "local",
+        select: locals.schema_version
+      )
 
-      {:ok, schema_version}
+    case Repo.one!(query, consistency: :one) do
+      schema_version ->
+        {:ok, schema_version}
     end
   end
 
   def execute_schema_change(conn, query) do
     result =
-      CSystem.run_with_schema_agreement(conn, fn ->
+      run_with_schema_agreement(conn, fn ->
         Xandra.execute(conn, query, %{}, consistency: :each_quorum, timeout: 60_000)
       end)
 
     case result do
       {:error, :timeout} ->
-        Xandra.Error.new(:agreement_timeout, "Schema agreement wait timeout.")
+        Xandra.Error.new(:agreement_timeout, "Schema agreement wait timeout.", [])
```

Comment:
if we're touching this, let's not use "private" (`@doc false`) functions

```suggestion
        %Xandra.Error{reason: :agreement_timeout, message: "Schema agreement wait timeout."}
```

still, I would make this change in a separate `fix(hk): crashes on schema change errors` commit

Reply:
I didn't want to touch that, but versions have been updated and raise warning

Reply:
well yeah `Xandra.Error.new` never actually accepted two parameters, if it did get there the process would just crash